src/dmd/toobj.d

```diff
@@ -840,25 +828,17 @@ void toObjFile(Dsymbol ds, bool multiobj)
                 }
                 else
                 {
-                    for (size_t i = 0; i < ti.members.dim; i++)
-                    {
-                        Dsymbol s = (*ti.members)[i];
-                        s.accept(this);
-                    }
+                    ti.members.foreachDsymbol( (s) { s.accept(this); } );
                 }
             }
         }
 
         override void visit(TemplateMixin tm)
         {
             //printf("TemplateMixin.toObjFile('%s')\n", tm.toChars());
-            if (!isError(tm) && tm.members)
+            if (!isError(tm))
             {
-                for (size_t i = 0; i < tm.members.dim; i++)
-                {
-                    Dsymbol s = (*tm.members)[i];
-                    s.accept(this);
-                }
+                tm.members.foreachDsymbol( (s) { s.accept(this); } );
             }
         }
```

Comment:
A good opportunity to change the code to:

```D
override void visit(TemplateMixin tm)
{
    if (isError(tm)) return;
    tm.members.foreachDsymbol( (s) { s.accept(this); } );
}
```

Reply:
Refactoring like this often reveals more refactoring that can be done. But I like to keep things incremental, because I often get stuck fixing regressions in other peoples' nnnnn line PRs.